### ISSUE TITLE
Block gallery: wrong tab order

### ISSUE THREAD

**Issue by nosolosw** [WordPress/gutenberg#14814]
When you tab through images within a gallery, the tab order is:

- image
- caption

Note that to access the remove control at the top right you have to focus backward. The tab order should instead be:

- remove control of the image selected
- caption

GIF:

![Peek 2019-04-04 10-14](https://user-images.githubusercontent.com/583546/55540177-b31a4c00-56c2-11e9-814c-e6220c4a0967.gif)

**Comment by nosolosw** [comment]
There is a proposal to fix this at https://github.com/WordPress/gutenberg/pull/14821

**Comment by nosolosw** [comment]
Copied over from https://github.com/WordPress/gutenberg/issues/14931 as per @afercia feedback:

---
Noticed while testing https://github.com/WordPress/gutenberg/pull/14822

In the gallery images, the order of the elements should be improved. Currently, the order is:
- "Remove Image" button (see `block-library-gallery-item__inline-menu`)
- the image (which is focusable)
- the caption field

With this order, there are at least 2 issues:
- when using a keyboard, the "Remove Image" button gets skipped because it appears _after_ the image is focused
- even if it wasn't skipped, the "Remove Image" button doesn't give any context on _which_ image it will remove because it's _before_ any relevant information related to the image

I'd propose to:
- move the "Remove Image" button after the image: shouldn't be a big problem as it's absolutely positioned
- make sure the button appears both when tabbing forward and when tabbing backwards

**Comment by afercia** [comment]
> move the "Remove Image" button after the image: shouldn't be a big problem as it's absolutely positioned

This would probably replicate the problem when tabbing backwards. I do realize rendering elements conditionally is very react-ish but sometimes that's not great for keyboard accessibility and tab order. For a similar issue with the Button component, see https://github.com/WordPress/gutenberg/pull/10128#issuecomment-484446630
